smart-contracts/contracts/cl-vault/src/query.rs

```diff
@@ -112,6 +113,15 @@ pub fn query_metadata(deps: Deps) -> ContractResult<MetadataResponse> {
     })
 }
 
+pub fn query_dex_router(deps: Deps) -> ContractResult<Option<String>> {
+    let dex_router = DEX_ROUTER.may_load(deps.storage)?;
```

Comment:
Could this be a .load()?; as the underlying state item is not an Option? Below code would be cleaner too.

Reply:
if we use load without dex router set then this query will error

Reply:
but in what case we would able to do so? the migrate() entrypoint sets this item at state. querying before migration is unfeasible. or there is a scenario where we migrate the code without executing the migrate() entrypoint that im missing?